homeassistant/helpers/intent.py

```diff
@@ -117,6 +124,60 @@ def __repr__(self):
         return '<{} - {}>'.format(self.__class__.__name__, self.intent_type)
 
 
+def fuzzyfinder(name, entities):
+    """Semi fuzzy matching function."""
+    matches = []
+    pattern = '.*?'.join(name)
+    regex = re.compile(pattern)
+    for entity in entities:
+        match = regex.search(entity)
+        if match:
+            matches.append((len(match.group()), match.start(), entity))
+    return [x for _, _, x in sorted(matches)]
+
+
+class ServiceIntentHandler(IntentHandler):
```

Comment:
Can you specify in both the name and doc of this class that it is only working for services that call entities and that it needs a name slot that will be mapped to `entity_id` in the service

Reply:
Updated. I think having the slot schema in the class also makes it clearer.